experimenter/tests/integration/nimbus/test_desktop_targeting.py

```diff
@@ -9,50 +11,101 @@
 
 
-@pytest.fixture(params=helpers.load_targeting_configs())
-def targeting_config_slug(request):
-    return request.param
-
-
 @pytest.fixture
+@cache
 def filter_expression_path():
     path = Path(__file__).parent / "utils" / "filter_expression.js"
     return path.absolute()
 
 
+@pytest.fixture(scope="function", autouse=True)
+def setup_browser(selenium, filter_expression_path):
+    """Open about:blank and initialize targeting environment once per test function."""
+    selenium.get("about:blank")
+
+    # Initialize TelemetryEnvironment and ExperimentAPI once
+    with filter_expression_path.open() as js:
+        result = Browser.execute_async_script(
+            selenium,
+            script=js.read(),
+            context="chrome",
+        )
+    assert result is True, "Failed to initialize targeting environment"
+
+    yield
+
+
 @pytest.mark.run_targeting
 def test_check_advanced_targeting(
     selenium,
-    targeting_config_slug,
     experiment_slug,
     default_data_api,
     filter_expression_path,
 ):
-    default_data_api["targetingConfigSlug"] = targeting_config_slug
-    experiment = helpers.create_experiment(
-        experiment_slug,
-        BaseExperimentApplications.FIREFOX_DESKTOP.value,
-        default_data_api,
-        targeting=targeting_config_slug,
-    )
-    logging.info(f"GraphQL creation: {experiment}")
-    experiment_data = helpers.load_experiment_data(experiment_slug)
-    targeting = experiment_data["data"]["experimentBySlug"]["jexlTargetingExpression"]
-    logging.info(f"Experiment Targeting: {targeting}")
-    recipe = experiment_data["data"]["experimentBySlug"]["recipeJson"]
-    logging.info(f"Experiment Recipe: {recipe}")
+    """
+    Evaluates all targeting configs in parallel using Promise.all for fast execution.
+    """
+    targeting_slugs = helpers.load_targeting_configs()
+    targeting_tests = []
 
-    # Inject filter expression
-    selenium.get("about:blank")
+    # Create experiments for all targeting configs
+    for idx, slug in enumerate(targeting_slugs):
+        test_slug = f"{experiment_slug}_{idx}_{slug}"
+        default_data_api["targetingConfigSlug"] = slug
+        helpers.create_experiment(
+            test_slug,
+            BaseExperimentApplications.FIREFOX_DESKTOP.value,
+            default_data_api,
+            targeting=slug,
+        )
+        logging.info(f"Created experiment: {test_slug}")
+        # Delay to prevent race conditions when creating many experiments
+        time.sleep(1)
+
+        experiment_data = helpers.load_experiment_data(test_slug)
+        targeting = experiment_data["data"]["experimentBySlug"]["jexlTargetingExpression"]
+        recipe = experiment_data["data"]["experimentBySlug"]["recipeJson"]
+
+        targeting_tests.append(
+            {
+                "slug": slug,
+                "targeting": targeting,
+                "recipe": json.dumps({"experiment": recipe}),
+            }
+        )
```

Comment:
We should not create any experiments and instead use a mapping of targeting slug to targeting expression. We may be able to just import `nimbus.targeting.constants` directly here 

This is the real bottleneck of this test.